### PR TITLE
fix(HACBS-1126): test suite for controllers/release is flaky

### PR DESCRIPTION
**Pull request by theflockers**
Fixes flaky release_adapter tests

- moves releasePlan/releaseAdmissionPlan creation to BeforeAll Block
- moves the test of unexistent releasePlan to the bottom

Signed-off-by: Leandro Mendes <lmendes@redhat.com>

### CODE REVIEW

File: controllers/release/release_adapter_test.go [hunk 1]
```diff
@@ -491,13 +472,27 @@ var _ = Describe("Release Adapter", Ordered, func() {
 		link, err := adapter.getReleasePlan()
 		Expect(err).Should(Succeed())
 		Expect(reflect.TypeOf(link)).To(Equal(reflect.TypeOf(&appstudiov1alpha1.ReleasePlan{})))
+	})
 
-		// It should err when the ReleasePlan does not exist
+	It("ensures the target context returns an error when no releaseplan exists", func() {
```
Comment (davidmogar) [inline controllers/release/release_adapter_test.go:477]: Can we use the change to  rewrite the description? Something like this:
```suggestion
	It("ensure an error is returned when setting the target context with a missing releaseplan", func() {
```

Comment (theflockers) [reply controllers/release/release_adapter_test.go:477]: sure.

**Review by davidmogar — APPROVED** [review]

**Comment by johnbieren** [comment]
Shouldn't we have just done the `Eventually` trick in the creation/deletion in the `BeforeEach` and `AfterEach` blocks instead? With things this way, we rely on each test not modifying the `ReleasePlan` and `ReleasePlanAdmission` or if they do, properly resetting it

**Comment by johnbieren** [comment]
@theflockers @davidmogar ping ^

**Comment by davidmogar** [comment]
I think we have some problems all over the tests that are actually difficult to see and that they appear randomly. I myself found one yesterday that I've been unable to replicate again to fix it. I also send you an IM about a problem I saw in one of the context tests. I'm kinda ignoring those problems because adding the `loader.go` file will cause multiple changes in the `release_adapter_test.go` file and that's a good opportunity to revisit this.

**Comment by johnbieren** [comment]
Sure, but if we create things in BeforeEach instead of BeforeAll, there is less chance that one test breaks another test. AIUI, once Leandro fixes the bug of the release can't be deleted if the release plan gets deleted first, I don't see any reason to not have things in BeforeEach

**Pull request merged**